lib/linter/linter.js

```diff
@@ -318,12 +336,16 @@ function createDisableDirectives(options) {
 }
 
 /**
- * Remove the ignored part from a given directive comment and trim it.
- * @param {string} value The comment text to strip.
- * @returns {string} The stripped text.
+ * Extract the directive and the justification from a given directive comment and trim them.
+ * @param {string} value The comment text to extract.
+ * @returns {{directivePart: string, justificationPart: string}} The extracted directive and justification.
  */
-function stripDirectiveComment(value) {
-    return value.split(/\s-{2,}\s/u)[0].trim();
+function extractDirectiveComment(value) {
+    const commentParts = value.split(/\s-{2,}\s/u);
+    const directive = commentParts[0].trim();
+    const justification = commentParts.length > 1 ? commentParts[1].trim() : "";
```

Comment:
If it happens that there are multiple ` -- `, this will treat only `commentParts[1]` as justification. Is it intentional?

For example:

```js
// eslint-disable-line no-undef -- foo -- bar
```

This returns:

```js
{
  directivePart: 'eslint-disable-line no-undef',
  justificationPart: 'foo'
}
```

While, since we are treating everything after the first `--` as directive's description, it might be expected to return:

```js
{
  directivePart: 'eslint-disable-line no-undef',
  justificationPart: 'foo -- bar'
}
```

Reply:
Fixed. Thanks.